packages/core/src/new-api/internal/validation/futures/validateNamedContractCall.ts

```diff
@@ -0,0 +1,53 @@
+import { ethers } from "ethers";
+
+import { IgnitionValidationError } from "../../../../errors";
+import { isArtifactType } from "../../../type-guards";
+import { ArtifactResolver } from "../../../types/artifact";
+import { NamedContractCallFuture } from "../../../types/module";
+
+export async function validateNamedContractCall(
+  future: NamedContractCallFuture<string, string>,
+  artifactLoader: ArtifactResolver
+) {
+  const artifact = await artifactLoader.load(future.contract.contractName);
```

Comment:
`future.contract` can may not be a named contract, but have the artifact inline instead.